cond.go

```diff
@@ -0,0 +1,253 @@
+package pgkit
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/Masterminds/squirrel"
+)
+
+type sqlExpr struct {
+	fn func() (string, []interface{}, error)
+}
+
+func (s sqlExpr) ToSql() (string, []interface{}, error) {
+	return s.fn()
+}
+
+func sqlExprFn(fn func() (string, []interface{}, error)) *sqlExpr {
+	return &sqlExpr{fn}
+}
+
+type condExpr struct {
+	op string
+
+	v interface{}
+}
+
+func (c condExpr) ToSql() (string, []interface{}, error) {
+	if c.v == nil {
+		return "", nil, nil
+	}
+
+	return "?", []interface{}{c.v}, nil
+}
+
+type condNode struct {
+	left  interface{}
+	right interface{}
+}
+
+func (n *condNode) ToSql() (string, []interface{}, error) {
+	_, leftIsString := n.left.(string)
+	_, rightIsExpr := n.right.(squirrel.Sqlizer)
+
+	if leftIsString && !rightIsExpr {
+		// backwards compatibility with old-style conditions
+		return squirrel.Eq{n.left.(string): n.right}.ToSql()
+	}
+
+	ql, argsl, err := compileLeaf(n.left)
+	if err != nil {
+		return "", nil, fmt.Errorf("error compiling left side: %v", err)
+	}
+
+	rl, argsr, err := compileLeaf(n.right)
+	if err != nil {
+		return "", nil, fmt.Errorf("error compiling right side: %v", err)
+	}
+
+	// always take operator from right side
+	if rl == "" {
+		return ql + compileOperator(n.right), argsl, nil
+	}
+
+	return string(ql) + compileOperator(n.right) + " " + string(rl), append(argsl, argsr...), nil
+}
+
+func compileNodes(nodes []squirrel.Sqlizer) (q string, args []interface{}, err error) {
+	for i, node := range nodes {
+		qn, argsn, err := node.ToSql()
+
+		if err != nil {
+			return "", nil, fmt.Errorf("error compiling node %d: %v", i, err)
+		}
```

Comment:
Removed line and `%v` -> `%w`
```suggestion
		qn, argsn, err := node.ToSql()
		if err != nil {
			return "", nil, fmt.Errorf("error compiling node %d: %w", i, err)
		}
```